test/cpp/jit/test_lite_interpreter.cpp

```diff
@@ -627,8 +629,119 @@ TEST(LiteInterpreterTest, GetByteCodeVersion) {
+  test_model_file_v5.append("script_module_v5.ptl");
+
   auto version_v4 = _get_model_bytecode_version(test_model_file_v4);
   AT_ASSERT(version_v4 == 4);
+
+  auto version_v5 = _get_model_bytecode_version(test_model_file_v5);
+  AT_ASSERT(version_v5 == 5);
+}
+
+TEST(LiteInterpreterTest, BackPortToVersionByteCodeModelV4ToV3) {
+  std::string filePath(__FILE__);
+  auto test_model_file_v4 =
+      filePath.substr(0, filePath.find_last_of("/\\") + 1);
+  test_model_file_v4.append("script_module_v4.ptl");
+
+  // Load check in model: script_module_v4.ptl
+  auto from_version = _get_model_bytecode_version(test_model_file_v4);
+  AT_ASSERT(from_version == 4);
+
+  // Backport script_module_v5.ptl to an older version
+  std::ostringstream oss;
+  const int64_t to_version_3 = 3;
+  bool backPortSuccess =
+      _backport_for_mobile(test_model_file_v4, oss, to_version_3);
+  AT_ASSERT(!backPortSuccess);
+}
+
+namespace {
+void runTorchScriptModel(Module m) {
+  // Run the backport model, then compare the result with expect
+  // result
+  auto input_data = std::vector<IValue>({IValue(1)});
+  auto actual_result = m.forward(input_data).toTensor();
+  auto expected_result = at::ones({2, 4}, ScalarType::Double) * 3;
+
+  AT_ASSERT(actual_result.equal(expected_result));
+}
+
+void runByteCodeModel(mobile::Module m) {
+  // Run the backport model, then compare the result with expect
+  // result
+  auto input_data = std::vector<IValue>({IValue(1)});
+  auto actual_result = m.forward(input_data).toTensor();
+  auto expected_result = at::ones({2, 4}, ScalarType::Double) * 3;
+
+  AT_ASSERT(actual_result.equal(expected_result));
+}
+
+void backportBasicCheck(
+    std::string test_model_file,
+    const int64_t expect_from_version) {
+  auto from_version = _get_model_bytecode_version(test_model_file);
+  AT_ASSERT(from_version == expect_from_version);
+
+  // Backport script_module_v5.ptl to an older version
+  constexpr int64_t minimum_to_version = 4;
+  int64_t current_to_version = from_version - 1;
+
+  AT_ASSERT(current_to_version >= minimum_to_version);
+
+  // Verify all candidate to_version work as expected.
+  while (current_to_version >= minimum_to_version) {
+    std::ostringstream oss;
+    bool backPortSuccess =
+        _backport_for_mobile(test_model_file, oss, current_to_version);
+    AT_ASSERT(backPortSuccess);
+
+    // Check backport model version
+    std::istringstream iss(oss.str());
+    auto backport_version = _get_model_bytecode_version(iss);
+    AT_ASSERT(backport_version == current_to_version);
+
+    // Load and run the backport model, then compare the result with expect
+    // result
+    mobile::Module m_mobile = _load_for_mobile(iss);
+    runByteCodeModel(m_mobile);
+
+    Module m_torch_script = load(iss);
+    runTorchScriptModel(m_torch_script);
+
+    current_to_version--;
+  }
+}
+} // namespace
+
+TEST(LiteInterpreterTest, BackPortToVersionByteCodeModelV5ToV4) {
```

Comment:
As discussed offline, please add a unit test from a current model to the backport version (now version 4). The original model is from a torchscript, something like
```
  torch::jit::Module m("m");
  // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
  m.register_parameter("weight", torch::ones({20, 1, 5, 5}), false);
  // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
  m.register_parameter("bias", torch::ones({20}), false);
  m.define(R"(
    def forward(self, input):
      x1 = torch.zeros(2, 2)
      x2 = torch.empty_like(torch.empty(2, 2))
      x3 = torch._convolution(input, self.weight, self.bias, [1, 1], [0, 0], [1, 1], False, [0, 0], 1, False, False, True, True)
      return (x1, x2, x3)
  )");
m.freeze();

// convert to mobile model m_mobile
bool backPortSuccess =
        _backport_for_mobile(test_model_file, oss, current_to_version);
// CHECK ...
```
So that for developers that bumps the bytecode version, this test will fail, unless the developer register the backport function for the version bump. After the backport function is registered, it would be covered by the same test without adding a new one. I think that what @raziel suggested on the test part.

Reply:
Yeah makes sense. I think I will have two test, one is from vn to the minimum version, and one is from vn  to minimum - 1 version, so we can check both success and failure case.